client/components/Colors.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import styled from 'styled-components';
+
+
+const ColorBox = styled.div`
+  float: left;
+  width: 50px;
+  height: 20px;
+  margin: 5px;
+  border: 1px solid rgba(0, 0, 0, .2);
+  cursor: pointer;
+`;
+
+const ColorName = styled.div`
+  font-size:10px;
+  margin: 5px;
+  clear: left;
+`;
+
+
+class Colors extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayColorName: false,
+      colorName: '',
+    };
+  }
+
+  onClick(color) {
+    this.changeColorName(color);
+    this.toggleColorName();
+  }
+
+  toggleColorName() {
+    this.setState({
+      displayColorName: !this.state.displayColorName,
+    });
+  }
+
+  changeColorName(color) {
+    this.setState({
+      colorName: color,
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        {
+          this.props.colors.map(color => (
+            <div>
+              <ColorBox style={{ backgroundColor: color }} onClick={() => this.onClick(color)}/>
```

Comment:
I believe we usually need to assign a key along with the mapping function. Toby shared an article last week that argued against simply using the array index. I'd take a look!